languages/java/oso/src/main/java/com/osohq/oso/Ffi.java

```diff
@@ -248,4 +295,23 @@ private Pointer checkResult(Pointer p) throws Exceptions.OsoException {
             return p;
         }
     }
+
+    private void processMessage(Pointer msgPtr) throws Exceptions.OsoException {
+        if (msgPtr == null) {
+            return;
+        }
+        String msgStr = msgPtr.getString(0);
```

Comment:
is this infallible?

Reply:
If you called this on something that wasn't actually a string you'd get some weird stuff happening depending on what it was pointing too. Likely you'd get some garbage which would then fail to parse as json below.
There's no exception to catch or anything, it'd be undefined bad stuff like any other ffi calls.